go/kbcrypto/util.go

```diff
@@ -6,12 +6,47 @@ package kbcrypto
 import (
 	"bytes"
 	"crypto/hmac"
+	"strings"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+const (
+	SCSigCannotVerify = int(keybase1.StatusCode_SCSigCannotVerify)
+	SCInvalidFormat   = int(keybase1.StatusCode_SCInvalidFormat)
+	SCBadFrame        = int(keybase1.StatusCode_SCBadFrame)
+	SCWrongType       = int(keybase1.StatusCode_SCWrongType)
+	SCNoKeyFound      = int(keybase1.StatusCode_SCNoKeyFound)
+	SCAPINetworkError = int(keybase1.StatusCode_SCAPINetworkError)
 )
 
+const wrongTypeStr = "Wrong saltpack message type: "
+const badFrameStr = "Error in framing: "
+
 func FastByteArrayEq(a, b []byte) bool {
 	return bytes.Equal(a, b)
 }
 
 func SecureByteArrayEq(a, b []byte) bool {
 	return hmac.Equal(a, b)
 }
+
+/*ParseVerificationOrDecryptionErrorForStatusCode takes an error string from a
+VerificationError or DecryptionError and returns a status code if there exists a match. */
+func ParseVerificationOrDecryptionErrorForStatusCode(msg string, defaultCode int, defaultName string) (code int, name string) {
```

Comment:
Why is it necessary to parse strings?

Reply:
Couldn't you just put this status code into libkb.DecryptionError when it happens?

Reply:
i removed the string parsing and added some type matching in go/engine/saltpack_decrypt.go, go/libkb/saltpack_dec.go, and go/engine/saltpack_verify.go. there are some errors that we won't be able to match on by type because those errors are just errorStrings, but it doesn't seem like we currently want to expose any of those errors to the frontend. lmk if this is looking ok.